packages/caliper-core/test/common/core/transaction-statisitcs-collector.js

```diff
@@ -0,0 +1,219 @@
+* limitations under the License.
+*/
+
+'use strict';
+
+const TransactionStatisticsCollector = require('../../../lib/common/core/transaction-statistics-collector');
+const TxStatus = require('../../../lib/common/core/transaction-status');
+
+const chai = require('chai');
+chai.should();
+const sinon = require('sinon');
+
+describe.only('TransactionStatisticsCollector implementation', () => {
+
+    let sandbox;
+    let clock;
+
+    beforeEach(function() {
+        sandbox = sinon.createSandbox();
+        clock = sandbox.useFakeTimers();
+    });
+
+    afterEach(function() {
+        sandbox.restore();
+    });
+
+    describe('Using a single TransactionStatisticsCollector to collect results', () => {
+
+        it('should not be possible to get basic round information from the collector', () => {
+            const transactionStatisticsCollector = new TransactionStatisticsCollector(0, 1, 'roundLabel');
+
+            transactionStatisticsCollector.getWorkerIndex().should.equal(0);
+            transactionStatisticsCollector.getRoundIndex().should.equal(1);
+            transactionStatisticsCollector.getRoundLabel().should.equal('roundLabel');
+        });
+
+        it('should not be possible to update a TransactionStatisticsCollector if it is not activated', () => {
+            const transactionStatisticsCollector = new TransactionStatisticsCollector();
+
+            const updateSpy = sandbox.spy(transactionStatisticsCollector, '_updateStatistics');
+
+            transactionStatisticsCollector.txSubmitted(1);
+            transactionStatisticsCollector.getTotalSubmittedTx().should.equal(0);
+
+            const txStatus = new TxStatus();
+            transactionStatisticsCollector.txFinished(txStatus);
+            sinon.assert.notCalled(updateSpy);
+        });
+
+        it('should be possible to update a TransactionStatisticsCollector with a single TxStatus results once activated', () => {
+            const transactionStatisticsCollector = new TransactionStatisticsCollector();
+            transactionStatisticsCollector.activate();
+
+            const updateSpy = sandbox.spy(transactionStatisticsCollector, '_updateStatistics');
+
+            transactionStatisticsCollector.txSubmitted(1);
+            transactionStatisticsCollector.getTotalSubmittedTx().should.equal(1);
+
+            clock.tick(50);
+            const txStatus = new TxStatus();
+            transactionStatisticsCollector.txFinished(txStatus);
+            sinon.assert.calledOnce(updateSpy);
+        });
+
+        it('should be possible to update a TransactionStatisticsCollector with an array of TxStatus results once activated', () => {
+            const transactionStatisticsCollector = new TransactionStatisticsCollector();
+            transactionStatisticsCollector.activate();
+
+            const updateSpy = sandbox.spy(transactionStatisticsCollector, '_updateStatistics');
+
+            transactionStatisticsCollector.txSubmitted(1);
```

Comment:
Do we need the submit part to test the finished part? (If yes, then 2 TXs should be submitted here, for consistency.)